src/transformers/models/llava/modeling_llava.py

```diff
@@ -46,6 +49,62 @@
 ]
 
 
+def get_anyres_image_grid_shape(image_size, grid_pinpoints, patch_size):
+    """
+    Calculate the shape of the image patch grid after the preprocessing for images of any resolution.
+
+    Args:
+        image_size (`tuple`):
+            The size of the input image in the format (width, height).
+        grid_pinpoints (`str`):
+            A string representation of a list of possible resolutions.
+        patch_size (`int`):
+            The size of each image patch.
+
+    Returns:
+        tuple: The shape of the image patch grid in the format (width, height).
+    """
+    if isinstance(grid_pinpoints, list):
+        possible_resolutions = grid_pinpoints
+    else:
+        possible_resolutions = literal_eval(grid_pinpoints)
```

Comment:
This is going to be called repeatedly in the forward pass. It's better to properly parse the grid points once in the model's init and then pass that to the method